### PR TITLE
util/topsql/reporter: migrate test-infra to testify

### LINKED ISSUE #26427: migrate test-infra to testify for `util/topsql/reporter` pkg

### PR DESCRIPTION
**Pull request by yedamao**
### What problem does this PR solve?

Issue Number: close #26427 

### Release note 

```release-note
None.
```

### CODE REVIEW

**Review by tisonkun — COMMENTED** [review]
Thanks for submitting this PR @yedamao ! Generally looks good. I'd like to know whether or not we run these tests in parallel.

**Comment by yedamao** [comment]
@tisonkun these tests depend some global variables in other pkg, we can not run these tests in parallel.

**Review by tisonkun — APPROVED** [review]
Thanks for your information @yedamao . LGTM.

I'd like to know which variables they depend on. May take a closer look to see if it is necessary. Global variable is general a bad idea.

**Comment by tisonkun** [comment]
/cc @xhebox @tiancaiamao @mmyj 

PTAL

File: util/topsql/reporter/reporter_test.go [hunk 1]
```diff
@@ -250,17 +237,17 @@ func (s *testTopSQLReporter) TestCollectAndTopN(c *C) {
...
-	c.Assert(results[2].SqlDigest, DeepEquals, []byte("sqlDigest3"))
-	c.Assert(getTotalCPUTime(results[2]), Equals, 3)
+	require.Nil(t, results[0].SqlDigest)
+	require.Equal(t, getTotalCPUTime(results[0]), 5)
+	require.Equal(t, results[0].RecordListTimestampSec, []uint64{0, 1, 3, 4})
+	require.Equal(t, results[0].RecordListCpuTimeMs, []uint32{1, 2, 1, 1})
+	require.Equal(t, results[1].SqlDigest, []byte("sqlDigest1"))
+	require.Equal(t, getTotalCPUTime(results[1]), 5)
+	require.Equal(t, results[2].SqlDigest, []byte("sqlDigest3"))
+	require.Equal(t, getTotalCPUTime(results[2]), 3)
```
Comment (xhebox) [inline util/topsql/reporter/reporter_test.go:247]: Reverted expect/target for these lines, should be `require.Equal(t, 3, getTotalCPUTime(results[2]))` or so.

Comment (yedamao) [reply util/topsql/reporter/reporter_test.go:247]: ok, I will fix it.

**Comment by yedamao** [comment]
@xhebox PTAL

**Review by xhebox — APPROVED** [review]

**Review by mmyj — COMMENTED** [review]
Could those test units be paralleled？

**Comment by yedamao** [comment]
Looks like these tests can't be paralleled. those tests will call setupRemoteTopSQLReporter to setup some global variables. 
```go
  variable.TopSQLVariable.MaxStatementCount.Store(int64(maxStatementsNum))
  variable.TopSQLVariable.ReportIntervalSeconds.Store(int64(interval))
  variable.TopSQLVariable.AgentAddress.Store(addr)
```

**Comment by tisonkun** [comment]
/cc @mmyj @crazycs520 

could you help on reviewing and merging this PR? It seems neat but too late get reviewed.

**Review by mmyj — APPROVED** [review]
lgtm

**Comment by mmyj** [comment]
> /cc @mmyj @crazycs520 
> 
> 
> 
> could you help on reviewing and merging this PR? It seems quite neat but too late get reviewed.

All done.👌

**Pull request merged**